arch/Kconfig

```diff
@@ -276,6 +276,12 @@ config FLASH_HAS_PAGE_LAYOUT
 	  This option is enabled when the SoC flash driver supports
 	  retrieving the layout of flash memory pages.
 
+config FLASH_HAS_DRIVER_ENABLED
```

Comment:
This should live in drivers/flash/Kconfig.  Not sure why FLASH_HAS_PAGE_LAYOUT doesn't live there as well.

Reply:
If I had moved this to flash-Kconfig, the Flash driver options were placed paraller to FLASH menuconfig, not under it as I expected.

Reply:
What do you mean parallel to it?

Reply:
![image](https://user-images.githubusercontent.com/17042885/37100871-67dc4fc8-2224-11e8-9dd3-b80d925f2f12.png)

Reply:
I found out that have to move the keword declaration before `menuconfig FLASH` for fix picture above.

Reply:
that makes sense, we should do that.

Reply:
done